src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -242,7 +242,7 @@ private bool MatchString(string str)
             {
                 while (c != 0)
                 {
-                    if (str[--c] != runtext![--pos])
+                    if (str[--c] != runtextSpan![--pos])
```

Comment:
can you remove the `!` null forgiving operator on a bunch of these now? Surprised that the compiler allows it on a struct.

Reply:
Oh good point, sure I'll fix that.

Reply:
@stephentoub  why did the compiler allow it?

Reply:
For the same reason it doesn't warn on `string s = "abc"; s!.GetHashCode();". The purpose of the operator is to make the null state of an expression be "not null". It doesn't warn if it already is "not null".

Reply:
Maybe something worth a minor cleanup analyzer.

Reply:
> Maybe something worth a minor cleanup analyzer.

https://github.com/dotnet/roslyn/issues/34714